### PR TITLE
Add object templates for functions

### PR DESCRIPTION
**Pull request by logicminds**
Previously pdk was not able to generate functions.  This adds the templates to do so.  This will need to merged before I can submit the PR in pdk to support functions.

### CODE REVIEW

File: object_templates/functions/function_spec.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+describe '<%= name %>' do
+  # please note that tese tests are examples only
```
Comment (sanfrancrisko) [inline object_templates/functions/function_spec.erb:4]: Typo:
```suggestion
  # please note that these tests are examples only
```

**Review by sanfrancrisko — DISMISSED** [review]
Just a minor typo, other than that LGTM. By chance, do you happen to have the PDK PR ready that this is associated with?

File: object_templates/functions/function_spec.erb (hunk 1 above)
Comment (logicminds) [reply object_templates/functions/function_spec.erb]: completed and rebased.

**Comment by logicminds** [comment]
@sanfrancrisko regarding the other PR.  It is linked here: puppetlabs/pdk#932  

It is complete and waiting to be merged.

**Review by DavidS — APPROVED** [review]

**Pull request merged**